pkg/xds/template/proxy_template_resolver.go

```diff
@@ -74,3 +74,14 @@ func (s sequentialResolver) GetTemplate(proxy *model.Proxy) *mesh_proto.ProxyTem
 func SequentialResolver(r ...ProxyTemplateResolver) ProxyTemplateResolver {
 	return sequentialResolver(r)
 }
+
+func SelectProxyTemplate(dataplane *core_mesh.DataplaneResource, proxyTemplates []*core_mesh.ProxyTemplateResource) *core_mesh.ProxyTemplateResource {
```

Comment:
That's good idea to extract the logic from `SimpleProxyTemplateResolver#GetTemplate` into separate function and use it to build `MatchedPolicies` object. But could you please refactor `SimpleProxyTemplateResolver#GetTemplate` method to use this new `SelectProxyTemplate` function to avoid code duplication?

Reply:
addressed